src/Sulu/Bundle/AdminBundle/Resources/js/components/CircleSelection/CircleSelectionRenderer.js

```diff
@@ -0,0 +1,196 @@
+            radius: radius * containerWidth,
+        };
+    }
+
+    componentDidMount() {
+        this.setInitialValue();
+    }
+
+    componentDidUpdate() {
+        this.setInitialValue();
+    }
+
+    setInitialValue = () => {
+        const {value} = this.props;
+
+        if (!value) {
+            this.handleChange(this.value);
+        }
+    };
+
+    static createNormalizers(props: Props): Array<Normalizer> {
+        const {containerWidth, containerHeight, maxRadius, minRadius, percentageValues, round, resizable} = props;
+
+        if (!containerWidth || !containerHeight) {
+            return [];
+        }
+
+        const normalizers = [
+            new PositionNormalizer(
+                containerWidth,
+                containerHeight
+            ),
+        ];
+
+        if (resizable) {
+            const calculatedMaxRadius = percentageValues && maxRadius ? maxRadius * containerWidth : maxRadius;
+            const calculatedMinRadius = percentageValues && minRadius ? minRadius * containerWidth : minRadius;
+
+            normalizers.push(
+                new SizeNormalizer(
+                    containerWidth,
+                    containerHeight,
+                    calculatedMaxRadius,
+                    calculatedMinRadius
+                )
+            );
+        }
+
+        if (round && !percentageValues) {
+            normalizers.push(new RoundingNormalizer());
+        }
+
+        return normalizers;
+    }
+
+    get normalizers() {
+        return CircleSelectionRenderer.createNormalizers(this.props);
+    }
+
+    normalize(selection: SelectionData): SelectionData {
+        return this.normalizers.reduce((data, normalizer) => normalizer.normalize(data), selection);
+    }
+
+    getMaximumSelection = (): SelectionData => {
+        const {containerWidth, containerHeight, resizable} = this.props;
+
+        let radius;
+        if (containerWidth && containerHeight && resizable) {
+            radius = Math.min(containerWidth, containerHeight) / 2;
+        } else {
+            radius = 0;
+        }
+
+        return this.normalize(
+            this.centerSelection({
+                left: 0,
+                top: 0,
+                radius,
+            })
+        );
+    };
+
+    centerSelection(selection: SelectionData): SelectionData {
+        const {containerWidth, containerHeight} = this.props;
+
+        const halfWidth = containerWidth / 2;
+        const halfHeight = containerHeight / 2;
+
+        return {
+            ...selection,
+            left: halfWidth,
+            top: halfHeight,
+        };
+    }
+
+    handleCircleDoubleClick = () => {
+        const {resizable} = this.props;
+
+        if (resizable) {
+            this.handleChange(this.getMaximumSelection());
+
+            return;
+        }
+
+        this.handleChange(this.normalize(this.centerSelection(this.value)));
+    };
+
+    handleCircleChange = (value: SelectionData) => {
```

Comment:
The `RectangleSelection` makes use of `MouseEvent.movementX` and `MouseEvent.movementY`, which is just the changeset. The `CircleSelection` on the other hand does some heavy computations which always return the full value, not just the changeset. So I would have to make additional computations to get just the changeset. And the performance issues were the same for the `CircleSelection` like for the `RectangleSelection`, but that should be fixed now.

Reply:
Okay, performance problems are fixed... And why can't we also use the `movementX` and `movementY` values for the `CircleSelection` but need heavy computations instead? (Especially the word heavy makes me a bit nervous, since this function is probably called many times).

Reply:
If you move the cursor like normal, `movementX` and `movementY` are very small, mostly `-1`, `0` or `1` ... And it's very inaccurate to make computations with those values. (Also I think it's not that heavy as it sounds 😄 but you can check that by yourself)

Reply:
Hm, okay... that being said, I still experience that the circle or rectangle moves faster than my cursor, which didn't happen (at least for the `RectangleSelection`) on the current `master` branch, and I somehow have the feeling that it is connected to this :thinking: